src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -50,115 +44,130 @@ public String getUrlForPage(String wikiName, String pageName) {
+  public String getUrlForWiki() {
+    return getUrlForWiki(Configuration.getWikiName(), Configuration.getLanguage(), false);
   }
 
-  public String getUrlForWiki() {
-    return getUrlForWiki(Configuration.getWikiName(), false);
+  public String getUrlForWiki(String wikiName, String language, boolean addWWW) {
+    return getUrlForWiki(wikiName, language, addWWW, envType);
   }
 
   public String getUrlForWiki(String wikiName) {
     return getUrlForWiki(wikiName, false);
   }
 
-  public String getUrlForWiki(String wikiName, EnvType envType) {
-    return getUrlForWiki(wikiName, false, envType);
-  }
-
-  public String getUrlForWiki(boolean addWWW) {
-    return getUrlForWiki(Configuration.getWikiName(), addWWW);
-  }
-
   public String getUrlForWiki(String wikiName, boolean addWWW) {
     return getUrlForWiki(wikiName, addWWW, envType);
   }
 
 
   public String getUrlForWiki(String wikiName, boolean addWWW, EnvType envType) {
+    return getUrlForWiki(wikiName, DEFAULT_LANGUAGE, addWWW, envType);
+  }
+
+  public String getUrlForWiki(String wikiName, String language) {
+    return getUrlForWiki(wikiName, language, false, envType);
+  }
+
+  public String getUrlForWiki(String wikiName, String language, boolean addWWW, EnvType envType) {
     final String wikiaName = getWikiaGlobalName(wikiName);
 
     String www = "";
     if (addWWW) {
       www = "www.";
     }
 
+    String host;
+    HttpUrl.Builder urlBuilder = new HttpUrl.Builder();
     switch (envType) {
       case DEV: {
         String devBoxOwner = this.env.split("-")[1];
-        return String.format(getUrlProtocol() + DEV_URL_FORMAT, www, wikiaName, devBoxOwner, envType.getWikiaDomain());
+        host = String.join(".", www + wikiaName, devBoxOwner, envType.getWikiaDomain());
+        break;
       }
       case PROD: {
-        return String.format(getUrlProtocol() + PROD_URL_FORMAT, www, wikiaName, envType.getWikiaDomain());
+        host = String.join(".", www + wikiaName, envType.getWikiaDomain());
+        break;
       }
       case STAGING: {
-        return String.format(getUrlProtocol() + PROD_URL_FORMAT, www, wikiaName, envType.getWikiaDomain());
+        host = String.join(".", www + wikiaName, envType.getWikiaDomain());
+        break;
       }
       case SANDBOX: {
-        if (newStagingUrlFormat) {
-          return String.format(getUrlProtocol() + SANDBOX_NEW_URL_FORMAT, www, wikiaName, this.env, envType.getWikiaDomain());
-        } else {
-          return String.format(getUrlProtocol() + SANDBOX_URL_FORMAT, this.env, www, wikiaName, envType.getWikiaDomain());
-        }
+        host = String.join(".", www + wikiaName, this.env, envType.getWikiaDomain());
+        break;
       }
       default:
-        return "";
+        throw new WebDriverException("Unknown environment type");
+    }
+    urlBuilder.scheme(getUrlProtocol()).host(host);
+
+    if (language != null && !("en").equals(language)) {
+      urlBuilder.addPathSegments(language);
```

Comment:
we still need to have an option to run with the old language format (as a subdomain)